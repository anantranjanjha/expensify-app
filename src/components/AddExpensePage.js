import React from "react"
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { AddExpenseHandle } from "../actions/expenses";


const AddExpensePAge = (props) => (
    <div>
        <ExpenseForm
            onSubmit={(expense) => {
                props.dispatch(AddExpenseHandle(expense));
                props.history.push('/');
            }
            }
        />
    </div>
);


export default connect()(AddExpensePAge);