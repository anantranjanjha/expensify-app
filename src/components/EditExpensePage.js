import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { StartEditExpenseHandle } from "../actions/expenses";
import { StartDeleteExpenseHandle } from "../actions/expenses";

const EditExpensePage = (props) => {
    return (
        <div>
            <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Edit Expense</h1>
            </div>
        </div>
            <div className="content-container">
            <ExpenseForm
                expense={props.expense}
                onSubmit={(expense) => {
                    props.dispatch(StartEditExpenseHandle(props.expense.id, expense));
                    props.history.push('/');
                }} />
            <button
            className="button button--secondary"
             onClick={(e) => {
                props.dispatch(StartDeleteExpenseHandle(props.expense.id))
                props.history.push('/');
            }}>Remove Expense</button>
            </div>
        </div>
    )
}

const mapstateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.description === props.match.params.id)
    };
};

export default connect(mapstateToProps)(EditExpensePage);