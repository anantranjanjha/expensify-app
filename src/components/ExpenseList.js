import React from "react";
import SelectorExpense from "../selectors/expenses";
import Item from "./Items";
import { connect } from "react-redux";

export const ExpenseList = (props) => (
    <div>
        <h1>ExpenseList</h1>
        {props.expenses.map((expense) => (
            <Item {...expense} key={expense.id} />
        ))}
    </div>
);

const mapstoreToProps = (state) => {
    return {
        expenses: SelectorExpense(state.expenses, state.filters),
    }
};

export default connect(mapstoreToProps)(ExpenseList);
