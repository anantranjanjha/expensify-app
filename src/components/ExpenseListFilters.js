import React, { Component } from "react";
import { DateRangePicker } from "react-dates";
import { setTextFilter, setStartDate, setEndDate } from "../actions/filters";
import { SortByAmount, SortByDate } from "../actions/filters";
import { connect } from "react-redux";

class ExpenseListFilters extends React.Component {

    state = {
        calenderfocused: null,
    };
    onDatesChangeHandle = ({ startDate, endDate }) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    }
    onFocusChange = (calenderfocused) => {
        this.setState(() => ({ calenderfocused }));
    };

    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input type="text"
                        className="text-input"
                        placeholder="Search Expenses"
                            value={this.props.filters.text}
                            onChange={(e) => {
                                this.props.dispatch(setTextFilter(e.target.value))
                            }}
                        />
                    </div>
                    <div className="input-group__item">
                        <select
                        className="select"
                            value={this.props.filters.sortBy}
                            onChange={(e) => {
                                (e.target.value === "date") ? this.props.dispatch(SortByDate()) : this.props.dispatch(SortByAmount());
                            }}>
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDateId="start_date_input"
                            endDateId="end_date_input"
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChangeHandle}
                            focusedInput={this.state.calenderfocused}
                            onFocusChange={this.onFocusChange}
                            numberOfMonths={1}
                            showClearDates={true}
                            isOutsideRange={() => false}
                        />
                    </div>
                </div>
            </div>
        );
    }
}


const mapStateToProps = (state) => {
    return {
        filters: state.filters,
    }
};

const ConnectedExpenseListFilters = connect(mapStateToProps)(ExpenseListFilters);

export default ConnectedExpenseListFilters;