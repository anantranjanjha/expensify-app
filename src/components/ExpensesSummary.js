import React, { useState } from "react";
import SelectorExpense from "../selectors/expenses";
import ExpenseTotal from "../selectors/expense-total";
import { connect } from "react-redux";
import numeral from "numeral";

const ExpensesSummary = (props) =>{
    const expenseTotal=numeral(ExpenseTotal(props.expenses)/100)
            .format('$0,0.00');
    return(
        <div>
        <h1>Viewing {props.expenses.length} expenses totalling {expenseTotal}</h1>
        </div>
    );
};

const mapstoreToProps = (state) => {
    return {
        expenses: SelectorExpense(state.expenses, state.filters),
    }
};

const ConnectedExpenseSummary=connect(mapstoreToProps)(ExpensesSummary)

export default ConnectedExpenseSummary;