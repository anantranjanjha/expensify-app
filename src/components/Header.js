import React from "react";
import { NavLink } from 'react-router-dom';
import { connect } from "react-redux";
import { startLogOut } from "../actions/auth";

const Header = ({startLogOut}) => (
    <header>
        <h1>Expensify App</h1>
        <NavLink to="/dashboard" activeClassName="is-active">Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">CreateExpense</NavLink>
        <button onClick={startLogOut}>LogOut</button>
    </header>
);

const mapDispatchToProps = (dispatch)=>({
    startLogOut: ()=> dispatch(startLogOut())
})


export default connect(undefined,mapDispatchToProps)(Header);