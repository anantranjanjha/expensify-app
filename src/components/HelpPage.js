import React from "react"

const HelpPage =() =>(
    <div>
    Help Page Showing
    </div>
)

export default HelpPage;