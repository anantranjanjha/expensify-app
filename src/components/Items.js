import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

export const Item = ({ description, amount, createdAt, id }) => (
    <Link className="list-item" to={`/edit/${description}`} >
        <div>
            <h3 className="list-item__title">{description}</h3>
            <span className="list-item__sub-title">{moment(createdAt).format('MMMM Do, YYYY')}</span>
        </div>
        <h3 className="list-item__data">{numeral(amount / 100).format('$0,0.00')}</h3>
    </Link>
);

const ConnectItem = connect()(Item);

export default ConnectItem;