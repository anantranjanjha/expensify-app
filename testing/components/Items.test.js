
import React from "react";
import { shallow } from "enzyme";
import { Item } from "../../src/components/Items";


const expense1 = {
    id: 1,
    description: 'debit',
    amount: 50000,
    note: 'gello',
    createdAt: 49,
};

test('testing React Router Component', () => {
    const wrapper = shallow(<Item expenses={expense1} />);
    expect(wrapper).toMatchSnapshot();
})