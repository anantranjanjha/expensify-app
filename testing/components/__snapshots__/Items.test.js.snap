// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing React Router Component 1`] = `
<div>
  <Link
    to="/edit/undefined"
  >
    <h1 />
  </Link>
  <p>
     - 
  </p>
</div>
`;
